bulk_site_creator/views.py

```diff
@@ -26,50 +37,145 @@
 @login_required
 @lti_role_required(const.ADMINISTRATOR)
 @lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
-@require_http_methods(['GET', 'POST'])
+@require_http_methods(["GET", "POST"])
 def index(request):
-    sis_account_id = request.LTI['custom_canvas_account_sis_id']
-    terms, _current_term_id = get_term_data_for_school(sis_account_id)
-    school_id = sis_account_id.split(':')[1]
-    school = School.objects.get(school_id=school_id)
+    sis_account_id = request.LTI["custom_canvas_account_sis_id"]
+    term_ids, _current_term_id_id = get_term_data_for_school(sis_account_id)
+    school_id = sis_account_id.split(":")[1]
     canvas_site_templates = get_canvas_site_templates_for_school(school_id)
     potential_course_sites_query = None
 
-    if request.method == 'POST':
-        selected_term = request.POST['course-term']
-        selected_template = request.POST['template-select']
-        # Retrieve all course instances for the given term and account that do not have Canvas course sites
+    if request.method == "POST":
+        selected_term_id = request.POST["course-term_id"]
+        # Retrieve all course instances for the given term_id and account that do not have Canvas course sites
         # nor are set to be fed into Canvas via the automated feed
-        potential_course_sites_query = get_course_instance_query_set(selected_term, sis_account_id).filter(
-            canvas_course_id__isnull=True, sync_to_canvas=0)
+        potential_course_sites_query = get_course_instance_query_set(
+            selected_term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True, sync_to_canvas=0)
 
     # TODO maybe better to use template tag unless used elsewhere?
-    potential_course_site_count = potential_course_sites_query.count() if potential_course_sites_query else 0
+    potential_course_site_count = (
+        potential_course_sites_query.count() if potential_course_sites_query else 0
+    )
 
     context = {
-        'terms': terms,
-        'potential_course_sites': potential_course_sites_query,
-        'potential_site_count': potential_course_site_count,
-        'canvas_site_templates': canvas_site_templates
-     }
-    return render(request, 'bulk_site_creator/index.html', context=context)
+        "term_ids": term_ids,
+        "potential_course_sites": potential_course_sites_query,
+        "potential_site_count": potential_course_site_count,
+        "canvas_site_templates": canvas_site_templates,
+    }
+    return render(request, "bulk_site_creator/index.html", context=context)
+
+
+@login_required
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["POST"])
+def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 61 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 122-180 into a separate function.